app/questionnaire/questionnaire_schema.py

```diff
@@ -562,6 +562,27 @@ def _update_dependencies_for_dynamic_options(
                 value_source, block_id=block_id, answer_id=answer_id
             )
 
+    def _update_dependencies_for_calculated_summary_value_source(
+        self, *, calculated_summary_id: str, block_id: str, answer_id: str | None
+    ) -> None:
+        """
+        For the given block (and optionally answer within) set it as a dependency of each calculated summary answer
+        If the calculated summary depends on a list, make the block depend on it too
+        """
+        # Type ignore: validator checks the validity of value sources.
+        calculated_summary_block: ImmutableDict = self.get_block(calculated_summary_id)  # type: ignore
+        answer_ids_for_block = get_calculated_summary_answer_ids(
+            calculated_summary_block
+        )
+        dependent = self._get_dependent_for_block_id(
+            block_id=block_id, answer_id=answer_id
+        )
+        for answer_id_for_block in answer_ids_for_block:
+            self._answer_dependencies_map[answer_id_for_block].add(dependent)
+            # if the answer is repeating, then the calculated summary also depends on the list it loops over
+            if list_name := self.get_list_name_for_answer_id(answer_id_for_block):
+                self._list_dependencies_map[list_name].add(dependent)
+
```

Comment:
Maybe not necessary as already explained what kind of answer ids they are:
```suggestion
        for answer_id in answer_ids_for_block:
            self._answer_dependencies_map[answer_id].add(dependent)
            # if the answer is repeating, then the calculated summary also depends on the list it loops over
            if list_name := self.get_list_name_for_answer_id(answer_id):
                self._list_dependencies_map[list_name].add(dependent)

```

Reply:
Yeah, I would have done this, but since there is an answer id being passed into the function I'm hesitant to overwrite it in the for loop as it could look like a mistake to someone revisiting this